_posts/2017-02-09-rapid-mysql-imports-docker.md

```diff
@@ -0,0 +1,186 @@
+
+Now, suppose that, instead of sequentially processing the SQL instructions contained in a database dump file, we were able to provide developers with a snapshot of the `/var/lib/mysql` directory for a given Drupal site. Would this be faster than the traditional database import methods?
+
+The answer is, yes, very much so!
+
+### Two test cases
+
+The table below shows the results of two test cases, one using a 19 MB database file and the other using a 4.7 GB database.
+
+| Method                | Database size | Time to drop tables and restore (seconds) |
+| :-------              | ------:       |                                  -------: |
+| Traditional `mysql`   | 19 MB         |                                       128 |
+| Docker volume restore | 19 MB         |                                        11 |
+| Traditional `mysql`   | 4.7 GB        |                                       606 |
+| Docker volume restore | 4.7 GB        |                                        85 |
+
+In other words, the MySQL data volume import completes, on average, in about 11% of the time that a traditional MySQL dump import would take!
+
+Since a GIF is worth a thousand words, compare these two processes side-by-side (both are using the same source 19 MB source database; the first is using a data volume restore process while the second is using the traditional MySQL import process):
+
+#### Data volume restore
+
+<script type="text/javascript" src="https://asciinema.org/a/102770.js" id="asciicast-102770" async data-autoplay="true" data-loop="true" data-theme="solarized-dark"></script>
+
+#### Traditional MySQL database dump import
+
+<script type="text/javascript" src="https://asciinema.org/a/102769.js" id="asciicast-102769" async data-autoplay="true" data-loop="true" data-theme="solarized-dark"></script>
+
+## Use MySQL volume for database imports with Docker
+
+Here's how the process works[^1]. Suppose you have a Docker stack with a web container and a database container, and that the database container has data in it already (your site is up and running locally). Assuming a database container name of `drupal_database`, to generate a volume for the MySQL `/var/lib/mysql` contents of the database container, you'd run these commands:
+
+``` bash
+docker stop drupal_database
+docker run --rm --volumes-from dupal_database carinamarina/backup backup \
+--source /var/lib/mysql/ --stdout --zip > db-data-volume.tar.gz
+```
+
+With the 4.7 GB sample database above, this process takes 239 seconds and results in 702 MB compressed file.
+
+We're making use of the [`carinamarina/backup`](https://getcarina.com/docs/tutorials/backup-restore-data/) image produced by Rackspace to create an archive of the database files.
+
+You can then distribute this file to your colleagues (we use Amazon S3), or make use of it in continuous integration builds (more on that below), using these commands:
```

Comment:
"(We use Amazon S3)" --> "(At Savas Labs, we use Amazon S3)"
And maybe include link for AWS.